### PR TITLE
[1.8.9] Update to latest BC + Thaumcraft API's, fix BC api trigger + actions

### PR DESCRIPTION
**Pull request by AlexIIL**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -24,4 +24,5 @@ bin/
 classes/
 .settings
 .metadata
-.project
\ No newline at end of file
+.project
+*.launch
```
Comment (AlexIIL) [inline .gitignore]: Eclipse gradle-generated launch config

**Comment by CovertJaguar** [comment]
The thaumcraft api is completely buggered atm, at least in my instance. Constantly switching branches has rendered it useless. Since 1.7 and 1.8 point at different repositories, I've lost the ability to check out 1.8 stuff. I've been considering splitting the Thaumcraft code into its own sub module.

**Comment by CovertJaguar** [comment]
Anyway looks good,  merged.

File: src/main/java/mods/railcraft/common/plugins/buildcraft/BuildcraftPlugin.java [hunk 2]
```diff
@@ -27,4 +37,12 @@ public static void addFacade(Block block, int meta) {
         FMLInterModComms.sendMessage("BuildCraft|Transport", "add-facade", String.format("%s@%d", GameData.getBlockRegistry().getNameForObject(block), meta));
     }
 
+    public static class EventHook {
+        @SubscribeEvent
+        @SideOnly(Side.CLIENT)
+        public static void textureStitchPre(TextureStitchEvent.Pre event) {
```
Comment (liach) [inline src/main/java/mods/railcraft/common/plugins/buildcraft/BuildcraftPlugin.java]: `static`?

**Pull request merged**